src/main/java/io/jmix/uisamples/view/flowui/components/virtuallist/inlineeditor/VirtualListInlineEditor.java

```diff
@@ -0,0 +1,195 @@
+    private static final String NOOP_DESCRIPTION = "";
+
+    @Autowired
+    private Resources resources;
+    @Autowired
+    private Messages messages;
+    @Autowired
+    private DialogWindows dialogWindows;
+
+    @ViewComponent
+    private DataContext dataContext;
+    @ViewComponent
+    private CollectionContainer<Food> foodDc;
+    @ViewComponent
+    private CollectionLoader<Food> foodDl;
+
+    @Subscribe
+    public void onBeforeShow(final BeforeShowEvent event) {
+        foodDl.load();
+    }
+
+    @Supply(to = "foodList", subject = "renderer")
+    private Renderer<Food> foodListRenderer() {
+        return new ComponentRenderer<>(item -> {
+            HorizontalLayout rootCardLayout = new HorizontalLayout();
+            rootCardLayout.setMargin(true);
+            rootCardLayout.setSpacing(true);
+
+            VerticalLayout infoLayout = new VerticalLayout();
+            infoLayout.setMargin(false);
+            infoLayout.setSpacing(false);
+            infoLayout.setPadding(false);
+            infoLayout.setWidth("30%");
+
+            Avatar avatar = new Avatar();
+            avatar.addThemeVariants(AvatarVariant.LUMO_XLARGE);
+
+            if(item.getIcon() != null && item.getIcon().length > 0) {
+                String iconName = "%s.png".formatted(item.getTitle());
+                avatar.setImageResource(new StreamResource(iconName, () -> new ByteArrayInputStream(item.getIcon())));
+            }
+
+            VerticalLayout verticalLayout = new VerticalLayout();
+            verticalLayout.setWidthFull();
+
+            HorizontalLayout itemDetailLayout = new HorizontalLayout();
+            itemDetailLayout.add(new Text(item.getDescription()));
+            itemDetailLayout.add(new Html(messages.formatMessage(getClass(), "foodListItemDescription", item.getPrice())));
+            itemDetailLayout.setPadding(false);
+            itemDetailLayout.setMargin(false);
+            itemDetailLayout.setAlignItems(FlexComponent.Alignment.CENTER);
+
+            infoLayout.add(new Html(messages.formatMessage(getClass(), "foodItemTitle", item.getTitle())));
+            infoLayout.add(itemDetailLayout);
+
+            VerticalLayout buttonsPanel = new VerticalLayout();
+            buttonsPanel.setWidth("AUTO");
+            buttonsPanel.setPadding(false);
+            buttonsPanel.setMargin(false);
+            buttonsPanel.setSpacing(false);
+
+            Button detailButton = new Button(new Icon(VaadinIcon.PENCIL));
+            detailButton.setText(messages.getMessage("actions.Edit"));
+            detailButton.addClickListener(e -> dialogWindows.detail(this, Food.class)
+                    .withViewClass(FoodDetailView.class)
+                    .editEntity(item)
+                    .withParentDataContext(getViewData().getDataContext())
+                    .withAfterCloseListener(closeEvent -> {
+                        FoodDetailView detail = closeEvent.getSource().getView();
```

Comment:
This variable is not used anywhere except in the `if` block. Why is it declared outside?